x/config/config.go

```diff
@@ -60,14 +61,62 @@ func WrapStreamDialer(dialer transport.StreamDialer, transportConfig string) (tr
 	}
 	var err error
 	for _, part := range strings.Split(transportConfig, "|") {
-		dialer, err = newStreamDialerFromPart(dialer, part)
+		dialer.StreamDialer, err = newStreamDialerFromPart(dialer.StreamDialer, part)
+		dialer.SanitizedConfig = dialer.SanitizeConfig(part)
 		if err != nil {
 			return nil, err
 		}
 	}
 	return dialer, nil
 }
 
+type StreamDialer struct {
+	transport.StreamDialer
+	SanitizedConfig string
+}
+
+const redactedPlaceholder = "REDACTED"
+
+// SanitizeConfig returns a sanitized version of the given config that could contain sensitive information
+func (sd *StreamDialer) SanitizeConfig(oneDialerConfig string) string {
```

Comment:
You know what. With the safer SanitizeConfig method, we might as well just expose that and revert all the changes to the dialer creation. It's also simpler because the same function applies to Packet and Stream.

Reply:
@fortuna That's the path taken in this PR #152. I also think so. The embedding of `sanitizedConfig` value in dialer touches too many parts of the code base. I still need to clean up PR #152 to have a `switch`/`case` and handle the unknown transport sanitization. I will update that shortly and seek your feedback. Thanks!